Automattic/jetpack
# JS Components: Add spacing props to Text component

This PR implements a spacing feature for the `<Text />` component.

The new props supported allow define the box-model for the component (margin and padding, not border)
Prop | Desc
------|------
`m` |  margin
`mt` | margin-top
`mr` | margin-right
`mb` | margin-bottom
`ml` | margin-left
`mx` | margin left and right
`my` | margin top and bottom
`p` |  padding
`pt` | padding-top
`pr` | padding-right
`pb` | padding-bottom
`pl` | padding-left
`px` | padding left and right
`py` | padding top and bottom

Example:
```jsx
function MyBoxModel() {
  // margin-left: 24px
  // margin-right: 24px
  // padding-bottom: 48px
  return (
    <Text mx={ 3 } pb={ 6 }>Hello</Text>
  );
}
```

Fixes #

#### Changes proposed in this Pull Request:

* JS Components: Add spacing props to Text component

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to My Jetpack overview
* Confirm the title of Plans and Connection sections are correct
  * font-size: 36px
  * margin-bottom: 24px;
<img width="431" alt="Screen Shot 2022-03-29 at 3 09 50 PM" src="https://user-images.githubusercontent.com/77539/160677310-0b0b702c-b077-402e-ae8c-91b19f1d10fa.png">

* Using Storybook
* Select the Box Model story of the Text component
* Check the spacing properties

https://user-images.githubusercontent.com/77539/160678313-b3b82fde-fc7c-472d-86ed-380402163f8e.mov
